src/array_api_typing/_namespace.py

```diff
@@ -0,0 +1,27 @@
+"""Static typing support for the array API standard."""
+
+__all__ = ["HasNamespace"]
+
+from typing import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class HasNamespace(Protocol):
```

Comment:
Do we want users to be able to use this as an abstract base class? Because if not, we should probably make it `@final` (making it a "pure structural type", instead of both a structural & nominal one)

Reply:
If you want to try playing around with that, that'd be great :)

Reply:
ah of course: https://github.com/python/mypy/issues/17288

Reply:
it's not very pretty, but personally I'd go with something like `# type: ignore[misc]  # see python/mypy#17288` then.

Reply:
SGTM. It's recognized as a bug so will hopefully eventually be fixed.

Reply:
Done.